docs/docs/code/nodejs/managing-state/README.md

```diff
@@ -0,0 +1,132 @@
+---
+short_description: Store and read data with the built in database.
+thumbnail: https://res.cloudinary.com/pipedreamin/image/upload/v1646763735/docs/icons/icons8-database-96_iv1oup.png
+---
+
+# Managing state
+
+In Node.js (Javascript) code steps, you can also store and retrieve data in code steps.
+
+This is very useful for tracking data between runs of a particular workflow.
+
+:::warning
+This functionality (`$.service.db`) is limited to only Node.js code steps at this time.
+
+Other step languages like [Python](/code/python/), [Bash](/code/bash/) and [Go](/code/go/) do not have this feature available yet.
+
+For more information on what functionality is available for those languages, please refer to their documentation.
+:::
+
+### Injecting the database
+
+By default, Node.js steps don't have access to the database service. It needs to be injected by defining it as a `prop`. 
+
+```javascript
+export default defineComponent({
+  props: {
+    // Define that the "db" variable in our component is a database
+    db: "$.service.db",
+  },
+  async run({ steps, $ }) {
+    // Now we can access the database at "this.db"
```

Comment:
Since this is the first time the user has seen `db.set` code, can you say explicitly what it's doing in the comment?